pkg/model/api/v1/role.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	modelAPI "github.com/perses/perses/pkg/model/api"
+)
+
+type RoleInterface interface {
+	GetMetadata() modelAPI.Metadata
+	GetRoleSpec() RoleSpec
+}
+
+type ActionKind string
+
+const (
+	KindWildcard ActionKind = "*"
+	KindRead     ActionKind = "read"
+	KindCreate   ActionKind = "create"
+	KindUpdate   ActionKind = "update"
+	KindDelete   ActionKind = "delete"
+)
+
+type Permission struct {
```

Comment:
we certainly want to check that `scopes` and `actions` are not empty (both of them)